src/components/Login/Login.tsx

```diff
@@ -19,49 +19,87 @@ const Login = ({ displayAsButtons = false }) => {
     provider: Moralis.Web3ProviderType = "metamask"
   ) => {
     event.preventDefault();
+    const id = toast.loading("Logging in...", { autoClose: 2000 });
+
     try {
       const user = await authenticate({ provider });
       if (user === undefined) {
         // user clicked "cancel" when prompted to sign message
         // @todo: update messaging
-        toast.error("You must sign the message to connect your wallet");
+        toast.update(id, {
+          render: "You must sign the message to connect your wallet",
+          type: "error",
+          isLoading: false,
+          autoClose: 2000,
+        });
         return;
       }
     } catch (error) {
       console.error(error);
       logUserOut();
-      toast.error(
-        "Something went wrong. Please refresh the page and try again."
-      );
+      toast.update(id, {
+        render: "Something went wrong. Please refresh the page and try again.",
+        type: "error",
+        isLoading: false,
+        autoClose: 2000,
+      });
       return;
     }
 
     try {
-      await connectWallet();
+      const connection = await connectWallet();
+      console.log(connection);
+      if (connection) {
+        toast.update(id, {
+          render: "Logged in",
+          type: "success",
+          isLoading: false,
+          autoClose: 2000,
+        });
+      } else {
+        toast.update(id, {
+          render: "Please register as warden",
+          type: "error",
+          isLoading: false,
+          autoClose: 2000,
+        });
+      }
```

Comment:
I think we can exclude this part. `connectWallet()` doesn't return anything. And it throws any errors that are encountered, so the `catch` block below will handle the different error states. It's a good idea to update the toast message when a user isn't registered though. To do this, I'd add another error type to `UserLoginError` enum and throw that from `connectWallet()` right before the user is redirected to the registration page. I'd give it the "info" type instead of "error" though. Although, I'm not sure if the toast messages persist when redirected. You'll have to test that.

Reply:
Actually, I tried to remove it and so, only the "logged in" pops.

Actually to manage this I added a final return statement to connectWallet : "connected" if the whole process is sucessfull, that's how I get it with :

```javascript
      if (connection) {
        toast.update(id, {
          render: "Logged in",
          type: "success",
          isLoading: false,
          autoClose: 2000,
        });
      } else {
        toast.update(id, {
          render: "Please register as warden",
          type: "error",
          isLoading: false,
          autoClose: 2000,
        });
      }
      ```
      
      But I can definitely remove the this block : 
      ```javascript 
      else {
        toast.update(id, {
          render: "Please register as warden",
          type: "error",
          isLoading: false,
          autoClose: 2000,
        });
      }
      ```
      As it won't persist upon navigation anyway